### PR TITLE
Create cosmos app and commands with scaffold tool

### PR DESCRIPTION
**Pull request by NicolasMahe**
Dependant on https://github.com/mesg-foundation/engine/pull/1640

todo:
- [x] rebase this PR on latest modif of https://github.com/mesg-foundation/engine/pull/1640

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,9 +54,9 @@ dep:
 build: check-version dep
 	go build -mod=readonly -o ./bin/engine -ldflags="-X 'github.com/mesg-foundation/engine/version.Version=$(version)'" core/main.go
 
-build-cmd-cosmos: dep
-	go build -mod=readonly -o ./bin/mesg-cosmos ./cmd/mesg-cosmos/main.go
-	go build -mod=readonly -o ./bin/mesg-cosmos-daemon ./cmd/mesg-cosmos-daemon/main.go
+build-cmd: dep
```
Comment (NicolasMahe) [inline Makefile]: let's not forget to rename this command in the phony stuff of make

File: cmd/mesg-cli/main.go [hunk 2]
```diff
@@ -53,28 +65,28 @@ func main() {
...
+		flags.LineBreak,
 		keys.Commands(),
+		flags.LineBreak,
 		version.Cmd,
 		flags.NewCompletionCmd(rootCmd, true),
 	)
 
-	executor := cli.PrepareMainCmd(rootCmd, "MESG", defaultCLIHome)
+	// Add flags and prefix all env exposed with AA
+	executor := cli.PrepareMainCmd(rootCmd, "AA", app.DefaultCLIHome)
```
Comment (NicolasMahe) [inline cmd/mesg-cli/main.go]:
```suggestion
	executor := cli.PrepareMainCmd(rootCmd, "MESG", app.DefaultCLIHome)
```

File: cmd/mesg-cli/main.go [hunk 3]
```diff
@@ -9,39 +10,50 @@ import (
...
-	basicManager = *enginesdk.NewBasicManager()
-	cdc := codec.Codec
+	// TODO: setup keybase, viper object, etc. to be passed into
+	// the below functions and eliminate global vars, like we do
+	// with the cdc
 
 	rootCmd := &cobra.Command{
-		Use:   "mesg-cosmos",
-		Short: "Cosmos Client",
+		Use:   "acli",
```
Comment (NicolasMahe) [inline cmd/mesg-cli/main.go]:
```suggestion
		Use:   "mesg-cli",
```

File: cmd/mesg-cli/main.go [hunk 4]
```diff
@@ -9,39 +10,50 @@ import (
...
 	if err != nil {
 		panic(err)
 	}
 	cosmos.CustomizeConfig(cfg)
 
-	basicManager = *enginesdk.NewBasicManager()
-	cdc := codec.Codec
+	// TODO: setup keybase, viper object, etc. to be passed into
+	// the below functions and eliminate global vars, like we do
+	// with the cdc
```
Comment (NicolasMahe) [inline cmd/mesg-cli/main.go]: remove this todo. it's a cosmos todo..

File: cmd/mesg-cli/main.go [hunk 5]
```diff
@@ -9,39 +10,50 @@ import (
...
 
 func main() {
-	// init app and codec
+	// Configure cobra to sort commands
+	cobra.EnableCommandSorting = false
+
+	// Instantiate the codec for the command line application
+	cdc := app.MakeCodec()
+
+	// Read in the configuration file for the sdk
```
Comment (NicolasMahe) [inline cmd/mesg-cli/main.go]:
```suggestion
	// Read in the configuration file for the sdk
	// TODO: change back when more refactor is done
```

File: cmd/mesg-daemon/main.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	cfg, err := config.New()
+	if err != nil {
+		panic(err)
+	}
+	cosmos.CustomizeConfig(cfg)
+
+	ctx := server.NewDefaultContext()
+	cobra.EnableCommandSorting = false
+	rootCmd := &cobra.Command{
+		Use:               "aud",
```
Comment (NicolasMahe) [inline cmd/mesg-daemon/main.go]:
```suggestion
		Use:               "mesg-daemon",
```

File: cmd/mesg-daemon/main.go [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+)
+
+const flagInvCheckPeriod = "inv-check-period"
+
+var invCheckPeriod uint
+
+func main() {
+	cdc := app.MakeCodec()
+
+	// config := sdk.GetConfig()
```
Comment (NicolasMahe) [inline cmd/mesg-daemon/main.go]:
```suggestion
	// TODO: put back when more refactor is done
	// config := sdk.GetConfig()
```

File: app/app.go [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+)
+
+const appName = "engine"
+
+var (
+	// DefaultCLIHome is the default home directories for the application CLI
+	DefaultCLIHome = os.ExpandEnv("$HOME/.mesg-cli")
+
+	// DefaultNodeHome sets the folder where the applcation data and configuration will be stored
+	DefaultNodeHome = os.ExpandEnv("$HOME/.mesg/tendermint")
```
Comment (NicolasMahe) [inline app/app.go]: this should be either fetch from engine config or removed from the engine config.

File: app/decorator.go [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type forceCheckTxProxyDecorator struct {
+	d sdk.AnteDecorator
+}
+
+func newForceCheckTxProxyDecorator(d sdk.AnteDecorator) forceCheckTxProxyDecorator {
```
Comment (NicolasMahe) [inline app/decorator.go]: i think this file is duplicated in `cosmos/decorator`.
let's remove the one in `cosmos/decorator`.

File: app/app.go [hunk 10]
```diff
@@ -0,0 +1,405 @@
...
+
+	// ModuleBasics The module BasicManager is in charge of setting up basic,
+	// non-dependant module elements, such as codec registration
+	// and genesis verification.
+	ModuleBasics = module.NewBasicManager(
+		genutil.AppModuleBasic{},
```
Comment (NicolasMahe) [inline app/app.go]: the params module is not present here:
```
params.AppModuleBasic{},
```
but i'm not sure it's useful. let's check the template and simapp to know

File: core/main.go [hunk 11]
```diff
@@ -148,10 +149,11 @@ func main() {
 	}
 
 	// init app
-	app, err := enginesdk.NewApp(tendermintLogger, db, cfg.Cosmos.MinGasPrices)
-	if err != nil {
-		logrus.WithField("module", "main").Fatalln(err)
-	}
+	// app, err := enginesdk.NewApp(tendermintLogger, db, cfg.Cosmos.MinGasPrices)
```
Comment (NicolasMahe) [inline core/main.go]: commented code to delete

File: core/main.go [hunk 12]
```diff
@@ -148,10 +149,11 @@ func main() {
...
+	initApp := app.NewInitApp(tendermintLogger, db, nil, true, 0, bam.SetMinGasPrices(cfg.Cosmos.MinGasPrices))
+	// if err != nil {
```
Comment (NicolasMahe) [inline core/main.go]: commented code to delete

File: app/app.go (hunk 8 above)
Comment (NicolasMahe) [reply app/app.go]: maybe let's just add a TODO comment for now.. i'm worry it will break stuff

File: app/app.go (hunk 10 above)
Comment (krhubert) [reply app/app.go]: It is used to create subspaces as well, let's keep it for now.

**Pull request merged**